File: geobus-backend/src/main/java/com/geobus/marrakech/service/StopService.java
package com.geobus.marrakech.service;

import java.util.List;

import com.geobus.marrakech.dto.StopDTO;
import com.geobus.marrakech.dto.TimeEstimationDTO;
import com.geobus.marrakech.model.Stop;

/**
 * Interface pour les services liés aux stations de bus
 */
public interface StopService {

    /**
     * Récupère toutes les stations de bus à Marrakech
     *
     * @return Liste des stations
     */
    List<StopDTO> getAllStopsInMarrakech();

    /**
     * Trouve la station la plus proche des coordonnées de l'utilisateur
     *
     * @param lat Latitude de l'utilisateur
     * @param lon Longitude de l'utilisateur
     * @return La station la plus proche avec distance et temps estimé
     */
    StopDTO getNearestStop(Double lat, Double lon);

    /**
     * Calcule le temps estimé pour atteindre une station spécifique
     *
     * @param userLat Latitude de l'utilisateur
     * @param userLon Longitude de l'utilisateur
     * @param stopId Identifiant de la station
     * @return Informations sur le temps de trajet
     */
    TimeEstimationDTO getTimeToStop(Double userLat, Double userLon, Long stopId);
}

File: geobusandroid/app/src/main/java/com/geobus/marrakech/api/StopService.kt
package com.geobus.marrakech.api

import com.geobus.marrakech.model.Stop
import com.geobus.marrakech.model.TimeEstimation
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Interface de service pour les requêtes liées aux stations de bus
 */
interface StopService {

    /**
     * Récupère toutes les stations de bus à Marrakech
     */
    @GET("stops/marrakech")
    suspend fun getAllStopsInMarrakech(): Response<List<Stop>>

    /**
     * Trouve la station la plus proche des coordonnées fournies
     */
    @GET("stops/nearest")
    suspend fun getNearestStop(
        @Query("lat") lat: Double,
        @Query("lon") lon: Double
    ): Response<Stop>

    /**
     * Calcule le temps estimé pour atteindre une station spécifique
     */
    @GET("stops/time")
    suspend fun getTimeToStop(
        @Query("userLat") userLat: Double,
        @Query("userLon") userLon: Double,
        @Query("stopId") stopId: Long
    ): Response<TimeEstimation>
}